deploy-agent/deployd/common/utils.py

```diff
@@ -99,8 +99,8 @@ def run_prereqs(config):
     # check if the puppet has finished or not
     if IS_PINTEREST:
         respect_puppet = config.respect_puppet()
-        if respect_puppet and not os.path.exists("/var/lib/puppet/state/state.yaml"):
-            log.info("Waiting for first successful puppet run.")
+        if respect_puppet and not os.path.exists(config.get_puppet_file_path()):
```

Comment:
To make it more defensive. You probably want to check if config.get_puppet_file_path()